### PR TITLE
Adding capability of multi-level inheritance to factory. Added tests

### PR DESCRIPTION
**Pull request by scalvert**
Added the ability for plugins to inherit from intermediate plugin. Will allow for better code reuse and sharing.

This PR paves the way for an intermediate modal plugin that can include code shared across Pinny and Pikabu. It should allow for those individual plugins to be much smaller.

Status: **Ready to merge**

Reviewers: @fractaltheory @donnielrt 

## Changes
- Updated the `Plugin.create` factory function to accept an extra parameter, the super constructor.
- Added tests.

## How to test-drive this PR
- checkout this branch
- `grunt test`
- marvel at its beauty

### CODE REVIEW

File: dist/plugin.js [hunk 1]
```diff
@@ -22,11 +22,16 @@
...
-        for (var key in Plugin.prototype) {
+    Plugin.create = function(name, SubConstructor, SuperConstructor, prototype) {
+        if (typeof prototype === 'undefined') {
+            prototype = SuperConstructor;
+            SuperConstructor = Plugin;
+        }
+
+        SubConstructor.__super__ = SuperConstructor;
+        for (var key in SuperConstructor.prototype) {
             if (!SubConstructor.prototype[key]) {
```
Comment (donnielrt) [inline dist/plugin.js]: do we not need a `hasOwnProperty` check here?

Comment (scalvert) [inline dist/plugin.js]: We can but it really doesn't matter, unless you can think of a use case where it would.

Comment (donnielrt) [inline dist/plugin.js]: I'd say let's add the check to be on the safe side, but wouldn't consider this a blocker.

**Comment by donnielrt** [comment]
LGTM :+1:

File: dist/plugin.js (hunk 1 above)
Comment (scalvert) [inline dist/plugin.js]: So I think adding the check would actually mess with the mechanism we're using for inheritance. Let's leave as is for now.

**Pull request merged**